src/Config/Loader/ClassLoader/LoggerLoader.php

```diff
@@ -184,7 +184,7 @@ public function resolveParent()
             $loggerNames = array_keys($this->instantiatedLoggers);
 
             while (strpos($name, '.') !== false) {
-                $current_parent = preg_replace('/\.[a-z]+$/', '', $name);
+                $current_parent = preg_replace('/\.[a-z]+$/i', '', $name);
```

Comment:
I was thinking about the complications here, and I think you should do one of the two following things.

Either replace the regex with `preg_replace('/\..+$/', '', $name);` so as to take away everything after the last `.` OR

```
// Store the pos so we only have to search through the string once
$last_delimit_pos = strrpos($name, '.');
while ($last_delimit_pos !== false) {
  $current_parent = substr($name, 0, $last_delimit_pos);
  $last_delimit_pos = strrpos($name, '.');
  // . . .
```

The reason is that your regex says to only remove things if there is a period, followed by ONLY letters until the end of the string. However, the rule we really care about is remove EVERYTHING regardless of what it is after the last delimiter, in this case a `.`.